### DISCUSSION TITLE
How to correctly use binding approach for updating UIs that are dependent on each other?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#2071] [Q&A] [answered]
### Question

Hi All,

I am creating few sliders where changing one will change the others, I need to build some separate logic for that. But I am trying to figure out how to use binding properly for this use case.

Ex: if I think of 2 sliders where both will sum up to 10, then if I use something like this below:
```python
class Demo:
    def __init__(self):
        self.number1 = 3
        self.number2 = 7

    def update_num1(self, num):
        self.number1 = num
        self.number2 = 10-self.number1

    def update_num2(self, num):
        self.number1 = 10-self.number1
        self.number2 = num
```
and then for the UIs I use
```python
sl1 = ui.slider(min=1, max=10, on_change=lambda e: demo.update_num1(e.value)).bind_value(demo, 'number1')
sl2 = ui.slider(min=1, max=10, on_change=lambda e: demo.update_num2(e.value)).bind_value(demo, 'number2')
```
This is causing an infinite loop, each one is pushing a change in the other one. How do I stop this?

Alternatively, I was using a normal function without binding which works okay:
```python
    def update_num(self):
        if self.number1 != sl1.value:
            self.number1 = sl1.value
            self.number2 = 10-self.number1
            sl2.value = self.number2
        if self.number2 != sl2.value:
            self.number2 = sl2.value
            self.number1 = 10 - self.number2
            sl1.value = self.number1
```

But I wanted to learn how to use binding properly in such cases, I think it would be more efficient than writing such functions every time. Can you please give some advice on this?

Thanks a lot,
Anindya

**Comment by falkoschindler** [comment, marked as answer]
Your update methods are incorrect. They should be
```py
    def update_num1(self, num):
        self.number1 = num
        self.number2 = 10 - self.number1

    def update_num2(self, num):
        self.number2 = num
        self.number1 = 10 - self.number2
```

But you could also implement the relation between both numbers as a property like this, removing the need for calling update methods at all:
```py
class Demo:
    def __init__(self):
        self.number1 = 3

    @property
    def number2(self):
        return 10 - self.number1

    @number2.setter
    def number2(self, value):
        self.number1 = 10 - value

demo = Demo()

ui.slider(min=1, max=10).bind_value(demo, 'number1')
ui.slider(min=1, max=10).bind_value(demo, 'number2')
```

**Reply by Anindya088** [reply]
Hi @falkoschindler ,

Thanks so much. That was stupid of me, that typo was causing the infinite loop 😢 
But the @property method that you showed is a far better solution, no need to use any update functions. Thanks for this, I will implement this.

Regards,
Anindya
